src/mod/transforms_maximal_overlap.jl

```diff
@@ -0,0 +1,102 @@
+
+"""
+Perform one level of the maximal overlap discrete wavelet transform (MODWT) on
+the the vector `v`, which is either the `j`th level scaling coefficients or,
+if `j==1`, the raw signal.  The vectors `h` and `g` are the MODWT detail and
+scaling filters.
+
+Returns a tuple `(v, w)` of the scaling and detail coefficients at level `j+1`.
+"""
+function modwt_step(v::Array{T,1}, j::Integer, h::Array{S,1},
+        g::Array{S,1}) where {T <: Number, S <: Number}
+    N = length(v)
+    L = length(h)
+    v1 = zeros(v)
+    w1 = zeros(v)
+    for t in 1:N
+        k = t
+        w1[t] = h[1] * v[k]
+        v1[t] = g[1] * v[k]
+        for n in 2:L
+            k -= 2^(j-1)
+            if k <= 0
+                k = mod1(k, N)
+            end
+            w1[t] += h[n] * v[k]
+            v1[t] += g[n] * v[k]
```

Comment:
The filters are reversed in `makereverseqmfpair`, but here the coefficients are computed as a convolution. I'm not sure, but I think for consistency with `dwt` you should either use ``makeqmfpair` and leave this as is, or compute the coefficients as a filtering (i.e. reversed convolution).

Reply:
I think this is a consequence of slightly different definitions/conventions in Percival and Walden vs. whatever reference(s) you used for the rest of the package...if you point me to them I can double-check.

Reply:
Another remark: on pp. 117-118 Percival and Walden give a correction to Daubechies' definition of LA(10), LA(14) and LA(18) scaling filters (order reversal to attain linear phase instead of the generalised linear phase).

Reply:
It is maybe time to review the mathematics of the current DWT implementation and write down the definition somewhere for myself and others.

Reply:
I'm changing the code in this PR to `makeqmfpair` for consistency for now, but will open an issue to discuss filter definitions.

Reply:
After some testing it looks like using the reversed filters was correct all along. It would be great if you could revert the change. There is still a difference in shift, but that can be fixed at a later time.

Reply:
Ha okay, no problem.